### DISCUSSION TITLE
Error in rclr..some variables became negative... Solver Tolerance ospsuite R

### DISCUSSION THREAD

**Discussion by DonkeySaddle** [Open-Systems-Pharmacology/Forum#1674] [Questions & Problems] [answered]
I am consistently getting this error and having to reduce the solver tolerances. Sometimes it works, sometimes I still get the error, sometimes it works for one dosage administration and then I get the error when I run the simulation for a different dose. 

Is there a more full proof way around this? I have looked through the discussions and haven't seen anything.

Thanks,

DS

**Comment by msevestre** [comment]
Which version are you using?

**Reply by DonkeySaddle** [reply]
ospsuite.utils_1.4.23
OSP11
R version 4.3.2

I am running a simulation with multiple drugs (though I am only administering 2 of them, just easier to be able to "turn" drugs on and off rather than build a new simulation for each possible interaction I want to examine)

**Reply by DonkeySaddle** [reply]
Mobi 11.2

**Reply by msevestre** [reply]
Yeah ok. We had an issue with 12.0 and I was wondering if you were using this.

**Reply by DonkeySaddle** [reply]
Didn't even know there was a 12.0, I downloaded it from [here](https://github.com/Open-Systems-Pharmacology/Suite/releases/tag/v11.2)

**Reply by msevestre** [reply]
12.0 is not released officially yet. We are beta testing it at the moment:)

**Reply by DonkeySaddle** [reply]
Looking forward to it!

**Comment by Yuri05** [comment, marked as answer]
What you could try:
* load your pkml simulation in MoBi
* open the MSV (Molecule Start Values) building block
* search for the molecule paths which are reported in the error message in R
* Check the "Neg. Values Allowed" checkbox
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/0b4e1247-3b68-4795-a946-832757786bc6)
* Update the MSV in your simulation from the MSV-building block
![grafik](https://github.com/Open-Systems-Pharmacology/Forum/assets/25061876/0fc53a04-7aef-4e6a-bc6d-334f41e465e2)
* Save modified simulation to pkml

You should also plot all quantities where you allowed negative values - to check, if the negative values are tolerable (can be considered as numerical noise) or not.

**Reply by DonkeySaddle** [reply]
Thanks @Yuri05 I bet this will do it. Appreciate the help. 

Tangentially related question, have you and the other developers considered an OSP package for Julia? It's fast and has excellent diffeq packages. I am probably in the minority, but being able to export simulations to Julia the way you can to Matlab would be amazing. 

Thanks,

DS

**Reply by StephanSchaller** [reply]
Well, we'd be happy to get sponsors to develop sth. like this :-)

**Reply by DonkeySaddle** [reply]
Something to aspire to!

**Reply by RobinM92** [reply]
Hi @Yuri05, 
Thanks for this solution. Do the MSV value settings remain when I then clone the simulation in pksim to simulate e.g. another dosing regimen? Or do we need to apply this workaround for every simulation scenario?

**Reply by Yuri05** [reply]
> Hi @Yuri05, Thanks for this solution. Do the MSV value settings remain when I then clone the simulation in pksim to simulate e.g. another dosing regimen? Or do we need to apply this workaround for every simulation scenario?

You mean, if you reimport then the modified simulation back into PK-Sim? In this case, you will not be able to clone the simulation.
So yes, unfortunately you will need to apply the workaround to every problematic simulation.

**Comment by Yuri05** [comment]
> Tangentially related question, have you and the other developers considered an OSP package for Julia? It's fast and has excellent diffeq packages. I am probably in the minority, but being able to export simulations to Julia the way you can to Matlab would be amazing.

Was under discussion some time ago, but no concrete plans so far.
If you are interested, please create a feature request.
